### PR TITLE
Add tests for creating #signatures

### PR DESCRIPTION
**Pull request by swoops**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [x] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

I added the following tests to cmd_zignature:
x86-64 function with hole
x86-64 function with jumpback
x86-64 r_sign_fcn_bytes bounds check
80386 function with hole
80386 function with jumpback
80386 r_sign_fcn_bytes bounds check
MIPS R3000 function with hole
MIPS R3000 function with jumpback
MIPS R3000  r_sign_fcn_bytes bounds check
PowerPC function with hole
PowerPC function with jumpback
PowerPC r_sign_fcn_bytes bounds check
ARM function with hole
ARM function with jumpback
ARM r_sign_fcn_bytes bounds check
ARM aarch64 function with hole
ARM aarch64 function with jmpback
ARM aarch64 r_sign_fcn_bytes bounds check

Mostly this checks edge cases when creating byte signatures. I manually double checked that the output was expected. I did not include any tests for sparc yet.

**Test plan**

Ensure the tests are relevant and useful and well named. Make sure I am not attempting to match too much information. 
I might of gotten carried away, feel free to tell me to reduce things.

All tests pass on my machine:
```
r2r db/cmd/cmd_zignature 
Running from ....test
Loaded 66 tests.
Skipping json tests because jq is not available.
[66/66]                    66 OK         0 BR        0 XX        0 FX
Finished in 12 seconds.
```

**Closing issues**
Does not close anything.

...

### CODE REVIEW

**Comment by XVilka** [comment]
Looks like some of the analysis commits has changed the outcome, could you please check the master and send a new pull request fixing those?

File: test/db/cmd/cmd_zignature [hunk 1]
```diff
@@ -1118,3 +1118,318 @@ EXPECT=<<EOF
...
+zaf
+z*
+EOF
+EXPECT=<<EOF
+zs *
+za sym.__aeabi_unwind_cpp_pr0 b 0030a0e3d4feffea
+za sym.__aeabi_unwind_cpp_pr0 g cc=27 nbbs=50 edges=75 ebbs=1 bbsum=1176
+za sym.__aeabi_unwind_cpp_pr0 o 0x0000183c
+za sym.__aeabi_unwind_cpp_pr0 r sym._Unwind_VRS_Get sym._Unwind_VRS_Get sym._Unwind_VRS_Set sym._Unwind_VRS_Get sym._Unwind_VRS_Get sym._Unwind_VRS_Set sym.__gnu_unwind_execute sym._Unwind_VRS_Get sym._Unwind_VRS_Set sym._Unwind_VRS_Set
+za sym.__aeabi_unwind_cpp_pr0 v s-48 s-20 s-36 s-24 s-16 s-15 s-12 s-68 s-8 s-44 s-40 s-52 s-56 s-4 s-32 r1 r0
```
Comment (ret2libc) [inline test/db/cmd/cmd_zignature:1365]: This made master branch red recently, probably because of recent changes in analysis that changed something in the identification of variables. Nonetheless, I think there is something wrong.

image

All variables in the function have `sp + X`, but here all variables are `s - Y`. Also, even though it says `s-84`, I don't see any variable in the function identified as `sp+0x54` nor `sp-0x54`. There seems to be something wrong.
